src/accelerate/accelerator.py

```diff
@@ -2580,6 +2580,9 @@ def save_model(
 
         os.makedirs(save_directory, exist_ok=True)
 
+        if any(param.device == torch.device("meta") for param in model.parameters()):
```

Comment:
Just a nit, how about moving this check before creating the directory, that way it's not created when saving fails.